### PR TITLE
simulator: the operator should share one limiter.

### LINKED ISSUE #5322: the leader should occupy one io limiter(100MB/s), not every operator.
## Development Task
refactor Move Operator cost:
1. all operator should share one limiter in one store 
2. Add Learner should generate and send snapshot rather  than AddPeer. 
3. some setting should be configurable.

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?

Issue Number: Close #5322

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

Code changes

Side effects

Related changes

### Release note

```release-note
 None.
```

### CODE REVIEW

File: tools/pd-simulator/simulator/task.go [hunk 1]
```diff
@@ -17,12 +17,36 @@ package simulator
...
 	"github.com/pingcap/kvproto/pkg/eraftpb"
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/tikv/pd/server/core"
 	"github.com/tikv/pd/tools/pd-analysis/analysis"
+	"github.com/tikv/pd/tools/pd-simulator/simulator/cases"
+)
+
+var (
+	chunkSize                = int64(4 * cases.KB)
```
Comment (lhy1024) [inline tools/pd-simulator/simulator/task.go:31]: units.KiB?

Comment (bufferflies) [reply tools/pd-simulator/simulator/task.go]: fix

**Comment by bufferflies** [comment]
PTAL @lhy1024 @HunDunDM again

File: tools/pd-simulator/simulator/task.go [hunk 2]
```diff
@@ -401,23 +425,38 @@ func (a *addLearner) IsFinished() bool {
...
 			n.stats.SendingSnapCount++
 		} else {
 			n.stats.ReceivingSnapCount++
 		}
 	}
-	stat.remainSize -= n.ioRate
-	// The sending or receiving process has not finished yet.
+
+	// store should generate/receive snapshot by chunk size.
+	for n.limiter.AllowN(int(chunkSize)) {
```
Comment (nolouch) [inline tools/pd-simulator/simulator/task.go:449]: It cannot be acceleration. we can configure the tick to sped up the simulator.

Comment (bufferflies) [reply tools/pd-simulator/simulator/task.go:450]: Fix it

File: tools/pd-simulator/simulator/task.go [hunk 3]
```diff
@@ -94,9 +114,22 @@ func responseToTask(resp *pdpb.RegionHeartbeatResponse, r *RaftEngine) Task {
...
+	action     snapAction
 	remainSize int64
-	finished   bool
+	status     snapStatus
+	start      time.Time
+}
+
+func newSnapshotState(size int64, action snapAction) *snapshotStat {
+	if action == Receive {
+		size = size / compressionRatio
```
Comment (HunDunDM) [inline tools/pd-simulator/simulator/task.go]: tiny fix

**Review by HunDunDM — APPROVED** [review]
rest LGTM

File: tools/pd-simulator/simulator/task.go (hunk 3 above)
Comment (bufferflies) [reply tools/pd-simulator/simulator/task.go]: fix

**Review by nolouch — APPROVED** [review]

**Pull request merged**